### ISSUE TITLE
User Profile : Add "Unselect all" action

### ISSUE THREAD

**Issue by Blandine-AA** [hhaccessibility/hhaccessibility.github.io#317]
In the Accessibility Requirements section, user can select all criteria at once, but cannot unselect them all.
It would be more user-friendly to allow the user to unselect all the selected criteria with 1 click. The "Select all" will toggle to "Unselect all" after clicking on it. Next click on "Unselect all" will unselect ALL the criteria.
Now, clicking again on select all simply remove the checkmark without action. I would expect it to unselect all the criteria.
![image](https://cloud.githubusercontent.com/assets/20052162/25434775/a0ffaec4-2a5b-11e7-90ae-ab0a976ad2dd.png)

**Comment by joshi1983** [comment]
I suggest having a "Select All" button and a separate "Clear All" button to separate the user's intent to change all checkboxes at once from reporting whether all, none, or some of the checkboxes are selected.

I think some users may be confused if a checkbox for "Select All" becomes checked and then says "Unselect all" which may be misread as meaning something like "all unselected" and make it harder to interpret the meaning of the checkmark next to the new label.

**Comment by joshi1983** [comment]
I like the removal of the checkbox.  What should the text be if there is 4 questions checked and 4 questions unchecked?

**Comment by Blandine-AA** [comment]
Good question.. Anytime there is a partial check, the text should be the default behavior "Select all". 

The other option would be to unselect all automatically without changing the text whenever the user clicks on the select all to remove the check mark. But I don't think that it is really user friendly. another option to think about !

**Comment by brucelau-github** [comment]
I will work on this issue

**Comment by joshi1983** [comment]
@brucelau-github I left a comment on https://github.com/hhaccessibility/hhaccessibility.github.io/pull/341.

The Ready for testing is mainly for after it has been merged and deployed so I removed that label since it is under code review right now.

**Comment by brucelau-github** [comment]
Okay

**Comment by joshi1983** [comment]
This issue can be tested by the following steps:
1. Sign in to view the profile.
1. Click the new "Select All" button.  See that all checkboxes under the same category gets checked and the caption change to "Unselect All".
1. Click the "Unselect All" button.  See that all the checkboxes get unchecked and the button caption returns to "Select All".
1. Click "Select All" and uncheck one of the checkboxes.  See that the button changes to say "Select All" again.  Any time at least 1 checkbox is unchecked, the caption should be "Select All".

@Blandine-AA did you want to test this?  You seem like the best person to verify that it matches what you wanted.

**Comment by Blandine-AA** [comment]
PASS
